### PR TITLE
Add python scripts for model prediction.

### PR DESCRIPTION
**Pull request by boru-roylu**
1. `predict.py` is the main file for prediction.
2. `prediction_helper.py` contains the functions that (1) gather the predictions from model output (2) write predictions into a file and (3) calculate recall@k.
3. `data/retrieval_dataloader.py` is dataloader for prediction. There are two mode. One is loading examples from pre-built image-text pairwise data. The other is loading image and text separately and creating pairs on the fly.

### CODE REVIEW

File: src/data/retrieval_dataloader.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+      index_keys = ['image_index', 'text_index', 'gt_image_index']
+      input_keys.extend(index_keys)
+      keep_feature_keys.extend(index_keys)
+
+    batch_size_per_replica = input_context.get_per_replica_batch_size(
+        config.global_batch_size)
+
+    drop_remainder = config.drop_remainder
+
+    if config.input_path != '':
```
Comment (santiaontanon-google) [inline src/data/retrieval_dataloader.py]: Isn't this equivalent to just "if config.input_path:"?

**Review by santiaontanon-google — APPROVED** [review]
LGTM

File: src/data/retrieval_dataloader.py (hunk 1 above)
Comment (jtainslie) [reply src/data/retrieval_dataloader.py]: +1 that Santi's suggestion would be more idiomatic.

File: src/data/retrieval_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+    # Gets the label of the image-text pair by matching image_index in
+    # image_dataset and gt_image_indx in text_dataset.
+    get_label = data_utils.get_retrieval_label_fn(self._params.pos_weight)
+    dataset = dataset.map(
+        get_label,
+        num_parallel_calls=tf.data.experimental.AUTOTUNE)
+
+    # Shards image pairs after enumerate all possible combinations.
+    if (input_context and input_context.num_input_pipelines > 1):
```
Comment (jtainslie) [inline src/data/retrieval_dataloader.py]: Nit: no need for parentheses.

File: src/predict.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+from absl import app
+from absl import flags
+from absl import logging
+
+import tensorflow as tf
+
+# Imports registered experiment configs.
+from official.core import exp_factory
+from official.core import task_factory
+from official.modeling.hyperparams import params_dict
```
Comment (jtainslie) [inline src/predict.py]: Nit: can be all one block.

File: src/prediction_helper.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+from absl import logging
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+
+from data import retrieval_dataloader
+from tasks import classification
+
+
+def get_recall_at_k_from_dataframe(df, topks=[1, 3, 5, 10]):
```
Comment (jtainslie) [inline src/prediction_helper.py]: Nit: the default value could be a tuple for immutability.
`topks=(1, 3, 5, 10)`

**Review by jtainslie — APPROVED** [review]
Readability lgtm.

**Review by wqfcr-google — APPROVED** [review]

**Pull request merged**